### PR TITLE
Feat(95): Read functions

### LINKED ISSUE #95: Implement `Read` functions for contracts & graph data
## Overview

Implementing a uniform data read approach for the several states through out the app is needed to streamline development

The most complex of these is the proposal state, which draws on the graph data feeds as well as on chain states.

## Acceptance criteria 

- [ ] Calls facilitated via hooks 
- [ ] Potentially would be good to make use of the recommended persistence approaches from wagmi  
- [ ] Graph calls facilitated via SSR server side endpoints 
- [ ] Any cached or persisted data should using `createStorage` the fetching and interaction of storage in this manner should be namespaced  

### Proposals

- [ ] Proposals should be fetched on load
- [ ] Proposals should make use of caching 
- [ ] Data should be updated regularly, the fastest would be per block

### Balances 

- [ ] Token balances should be fetched upon wallet connection 
- [ ] Data should be fetched on block [using query invalidation](https://wagmi.sh/react/guides/tanstack-query)
- [ ] A unified get balances hook should be used to fetch

### Locks 

- [ ] Querying locks should be fetched upon wallet connection 
- [ ] Querying locks by user should be done using graph endpoints

### PR DESCRIPTION
**Pull request by RyRy79261**
## Overview 

This PR refactors a lot of the data reads to make access more streamlined, `graphql-codegen` has been used to create the various hooks with wrappers being added to join chain & graph data together.

## Testing 

This ticket simply refactors how we collect and manage data, styling is handled in #94 

Look through the app to see if data is pulling as expected

## Related tickets

- closes #95

### CODE REVIEW

File: src/lib/contracts/locking/useLockingTotalSupply.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useContracts } from "@/lib/contracts/useContracts";
+import { useReadContract } from "wagmi";
+import { LockingABI } from "@/lib/abi/Locking";
+
+const useLockingTotalSupply = () => {
```
Comment (denviljclarke) [inline src/lib/contracts/locking/useLockingTotalSupply.ts:5]: Like on the FE, I think referring to Locking as veMento is a bit clearer e.g `useVeMentoTotalSupply`

File: src/lib/contracts/useGetContractsInfo.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  [address: string]: {
+    abi?: string;
+    name?: string;
+  };
+};
+
+const useGetContractsInfo = ({ calls }: Props) => {
+  const chainId = useChainId();
+
+  //   const { data, error: apolloError } = useQuery(GetContractsInfo, {
```
Comment (denviljclarke) [inline src/lib/contracts/useGetContractsInfo.ts:100]: Do we still need this?

File: src/lib/graphql/celo-explorer/generated/celoGraph.tsx [hunk 3]
```diff
@@ -1,65 +1,28 @@
 /* eslint-disable */
-import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
+import { gql } from '@apollo/client';
```
Comment (denviljclarke) [inline src/lib/graphql/celo-explorer/generated/celoGraph.tsx:2]: Was this generated as a tsx file? It feels like it should be .ts

File: codegen.ts [hunk 4]
```diff
@@ -12,18 +12,50 @@ const config: CodegenConfig = {
   generates: {
     // NOTE: In case we need to use different subgraph URLs for different environments
     // we'll need to add another element to the object below, i.e. "./app/graphql/subgraph-alfajores/generated"
-    "./src/lib/graphql/subgraph/generated/": {
+    "./src/lib/graphql/subgraph/generated/subgraph.tsx": {
```
Comment (denviljclarke) [inline codegen.ts:15]: Ahh, the output file is specified here. Let's change this to .ts. wdyt?

File: src/lib/contracts/useGetContractsInfo.ts (hunk 2 above)
Comment (RyRy79261) [reply src/lib/contracts/useGetContractsInfo.ts]: Its a read query in used in the app, do you mean the comment?

File: src/lib/contracts/locking/useLockingTotalSupply.ts (hunk 1 above)
Comment (RyRy79261) [reply src/lib/contracts/locking/useLockingTotalSupply.ts]: I've nuked it after the fix PR #102

File: codegen.ts (hunk 4 above)
Comment (denviljclarke) [reply codegen.ts:15]: Ok np, got it. It just felt weird at a glance 😅

File: src/lib/contracts/useGetContractsInfo.ts (hunk 2 above)
Comment (denviljclarke) [reply src/lib/contracts/useGetContractsInfo.ts]: Yeh, the comment. It seems I selected the wrong line. Looks good now 👍

**Comment by denviljclarke** [comment]
LGTM 🫡

File: src/app/(routes)/proposals/[id]/page.tsx [hunk 5]
```diff
@@ -1,20 +1,17 @@
 "use client";
-import { useMemo } from "react";
```
Comment (chapati23) [inline src/app/(routes)/proposals/[id]/page.tsx:2]: Details page is broken for me, somewhere we're trying to serialize a BigInt

https://governance-2esaba9d7-mentolabs.vercel.app/proposals/1788130690851969321761660844276733770864782080735732005121452579323831102760

image

**Review by chapati23 — CHANGES_REQUESTED** [review]
Details page is broken for me

File: src/app/(routes)/proposals/[id]/page.tsx (hunk 5 above)
Comment (RyRy79261) [reply src/app/(routes)/proposals/[id]/page.tsx:2]: It's broken on dev as well 🤔

**Review by chapati23 — APPROVED** [review]
fixed a small regression with the wrong execution code component being used, now all read functions look good AFAICS!

**Pull request merged**